<a href="https://colab.research.google.com/github/thejustinfields/project2_/blob/main/Project2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#First choice: Spanish Wine Quality

1. Source of data
- fedesoriano. (April 2022). Spanish Wine Quality Dataset. Retrieved [May 11, 2023] from https://www.kaggle.com/datasets/fedesoriano/spanish-wine-quality-dataset


In [ ]:
import pandas as pd
path = "/content/drive/MyDrive/wines_SPA.csv"
df = pd.read_csv(path)
df.head()

winery           wine  year  rating  num_reviews country  \
0  Teso La Monja          Tinto  2013     4.9           58  Espana   
1         Artadi  Vina El Pison  2018     4.9           31  Espana   
2   Vega Sicilia          Unico  2009     4.8         1793  Espana   
3   Vega Sicilia          Unico  1999     4.8         1705  Espana   
4   Vega Sicilia          Unico  1996     4.8         1309  Espana   

             region   price                  type  body  acidity  
0              Toro  995.00              Toro Red   5.0      3.0  
1    Vino de Espana  313.50           Tempranillo   4.0      2.0  
2  Ribera del Duero  324.95  Ribera Del Duero Red   5.0      3.0  
3  Ribera del Duero  692.96  Ribera Del Duero Red   5.0      3.0  
4  Ribera del Duero  778.06  Ribera Del Duero Red   5.0      3.0

2. Brief description of data
- The dataset looks at spanish wine and how the relationship of its popularity and description effect the quality.

3. What is the target?
- The target is "rating".

4. What does one row represent? (A person?  A business?  An event? A product?)
- One row represents a specific spanish wine

5. Is this a classification or regression problem?
- This is a regression problem

6. How many features does the data have?
- The data has 11 features.

7. How many rows are in the dataset?
- There are 7500 rows in the dataset

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7500 entries, 0 to 7499
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   winery       7500 non-null   object 
 1   wine         7500 non-null   object 
 2   year         7498 non-null   object 
 3   rating       7500 non-null   float64
 4   num_reviews  7500 non-null   int64  
 5   country      7500 non-null   object 
 6   region       7500 non-null   object 
 7   price        7500 non-null   float64
 8   type         6955 non-null   object 
 9   body         6331 non-null   float64
 10  acidity      6331 non-null   float64
dtypes: float64(4), int64(1), object(6)
memory usage: 644.7+ KB


In [ ]:
df.value_counts()

winery                            wine                   year  rating  num_reviews  country  region            price  type                  body  acidity
Contino                           Rioja Graciano         2011  4.2     395          Espana   Rioja             61.94  Rioja Red             4.0   3.0        199
Artadi                            Valdegines             2016  4.2     398          Espana   Vino de Espana    48.96  Tempranillo           4.0   2.0        199
Sierra Cantabria                  Coleccion Privada      2012  4.2     420          Espana   Rioja             28.30  Rioja Red             4.0   3.0        199
La Rioja Alta                     Vina Alberdi Crianza   2018  4.2     410          Espana   Rioja             13.58  Rioja Red             4.0   3.0        199
Bodegas La Horra                  Corimbo I              2015  4.2     412          Espana   Ribera del Duero  37.90  Ribera Del Duero Red  5.0   3.0        199
                                         

In [ ]:
df.duplicated

<bound method DataFrame.duplicated of                     winery             wine  year  rating  num_reviews  \
0            Teso La Monja            Tinto  2013     4.9           58   
1                   Artadi    Vina El Pison  2018     4.9           31   
2             Vega Sicilia            Unico  2009     4.8         1793   
3             Vega Sicilia            Unico  1999     4.8         1705   
4             Vega Sicilia            Unico  1996     4.8         1309   
...                    ...              ...   ...     ...          ...   
7495               Contino          Reserva  2016     4.2          392   
7496  Conreria d'Scala Dei    Les Brugueres  2018     4.2          390   
7497           Mustiguillo  Finca Terrerazo  2017     4.2          390   
7498           Matarromera     Gran Reserva  2011     4.2          389   
7499              Sei Solo         Preludio  2016     4.2          388   

     country            region   price                  type  body  acidi

#Second choice: Adult Income

1. Source of data
- https://www.kaggle.com/datasets/wenruliu/adult-income-dataset

2. Brief description of data
- The dataset looks at the factors that effect an individual's annual income.



In [ ]:
path_2 = "/content/drive/MyDrive/adult.csv"
df_2 = pd.read_csv(path_2)
df_2.head()


age  workclass  fnlwgt     education  educational-num      marital-status  \
0   25    Private  226802          11th                7       Never-married   
1   38    Private   89814       HS-grad                9  Married-civ-spouse   
2   28  Local-gov  336951    Assoc-acdm               12  Married-civ-spouse   
3   44    Private  160323  Some-college               10  Married-civ-spouse   
4   18          ?  103497  Some-college               10       Never-married   

          occupation relationship   race  gender  capital-gain  capital-loss  \
0  Machine-op-inspct    Own-child  Black    Male             0             0   
1    Farming-fishing      Husband  White    Male             0             0   
2    Protective-serv      Husband  White    Male             0             0   
3  Machine-op-inspct      Husband  Black    Male          7688             0   
4                  ?    Own-child  White  Female             0             0   

   hours-per-week native-country income  
0              40  United-States  <=50K  
1              50  United-States  <=50K  
2              40  United-States   >50K  
3              40  United-States   >50K  
4              30  United-States  <=50K

3. What is the target?
- The target is "income"

4. What does one row represent? (A person?  A business?  An event? A product?)
- One row represents an individual

5. Is this a classification or regression problem?
- This is a classification problem

6. How many features does the data have?
- The data has 15 features

In [ ]:
df_2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48842 entries, 0 to 48841
Data columns (total 15 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   age              48842 non-null  int64 
 1   workclass        48842 non-null  object
 2   fnlwgt           48842 non-null  int64 
 3   education        48842 non-null  object
 4   educational-num  48842 non-null  int64 
 5   marital-status   48842 non-null  object
 6   occupation       48842 non-null  object
 7   relationship     48842 non-null  object
 8   race             48842 non-null  object
 9   gender           48842 non-null  object
 10  capital-gain     48842 non-null  int64 
 11  capital-loss     48842 non-null  int64 
 12  hours-per-week   48842 non-null  int64 
 13  native-country   48842 non-null  object
 14  income           48842 non-null  object
dtypes: int64(6), object(9)
memory usage: 5.6+ MB


7. How many rows are in the dataset?
- There are 48,842 rows

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?